dbms/src/Flash/Coprocessor/DAGExpressionAnalyzer.cpp

```diff
@@ -1133,6 +1133,36 @@ String DAGExpressionAnalyzer::appendNullAwareSemiJoinEqColumn(
     return and_arg_names.size() == 1 ? and_arg_names[0] : applyFunction("and", and_arg_names, last_step.actions, nullptr);
 }
 
+bool DAGExpressionAnalyzer::appendRuntimeFilterProperties(RuntimeFilterPtr & runtime_filter)
+{
+    NameAndTypePair name_and_type;
+    if (!isColumnExpr(runtime_filter->getSourceExpr()))
+    {
+        return false;
+    }
+    name_and_type = getColumnNameAndTypeForColumnExpr(runtime_filter->getSourceExpr(), getCurrentInputColumns());
+    runtime_filter->setSourceColumnName(name_and_type.name);
+    Block header;
+    std::shared_ptr<Set> in_values_set;
+    const auto & settings = context.getSettingsRef();
+    switch (runtime_filter->getRFType())
+    {
+    case tipb::IN:
+        in_values_set = std::make_shared<Set>(
+            SizeLimits(settings.rf_max_in_value_set, settings.max_bytes_in_set, settings.set_overflow_mode),
+            TiDB::TiDBCollators{getCollatorFromExpr(runtime_filter->getSourceExpr())});
```

Comment:
Is different collator handled properly for string column in the storage? Because for different collator, the min/max value of the string column should be different, I'm not sure rough set index support it or not.

Reply:
The function ```getCollatorFromExpr``` is used to get different collector by different type

Reply:
I forbidden unsupported type in ```RuntimeFilter::build```